announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -3131,6 +3178,97 @@ public void doDelete_announcement_link(RunData rundata, Context context)
 			state.setStatus(DELETE_ANNOUNCEMENT_STATUS);
 		}
 	} // doDelete_announcement_link
+ 
+	private void loadSelectedAnnouncements(RunData rundata, Context context) {
+
+		// retrieve the state from state object
+		AnnouncementActionState state = (AnnouncementActionState) getState(context, rundata, AnnouncementActionState.class);
+
+		String peid = ((JetspeedRunData) rundata).getJs_peid();
+		SessionState sstate = ((JetspeedRunData) rundata).getPortletSessionState(peid);
+
+		// get the channel and message id information from state object
+		String messageReference = state.getMessageReference();
+
+		// then, read in the selected announcment items
+		String[] messageReferences = rundata.getParameters().getStrings("selectedMembers");
+		Vector v = new Vector();
+
+		if (messageReferences == null) state.setDeleteMessages(v);
+
+		for (int i = 0; i < messageReferences.length; i++) {
+			// get the message object through service
+			try {
```

Comment:
Can we directly check for messageReference null then do enhanced for-loop and remove the check in here for `setDeleteMessages(v)` do it at the end.